source/portable/os/ota_os_posix.c

```diff
@@ -402,13 +409,19 @@ OtaErr_t Posix_OtaDeleteTimer( OtaTimerId_t otaTimerId )
 void * STDC_Malloc( size_t size )
 {
     /* Use standard C malloc.*/
-    /* coverity[misra_c_2012_rule_21_3_violation] This is on linux. */
+
+    /* MISRA rule 21.3 prohibits the use of malloc and free from stdlib.h. But this implementation
+     * is on linux, so it's a false positive. */
```

Comment:
I understand that use of malloc is ubiquitous on systems with virtual memory, but I don't see how that makes it a false positive.  There are three listed examples of undefined behavior in the rationale section of rule 21.3.  I think it would be better to justify the deviation by explaining why these behaviors do not pertain.

Reply:
Because this implementation only run on linux desktop, it's not for embedded devices running RTOS. I don't see why we're not allowed to use malloc here, otherwise how would you recommend implementing the malloc interface on linux?

Reply:
I'd say, don't use dynamic allocation at all.  Barring that, I'd say use malloc, but explain how you use is safe (CBMC proofs?).  Also, wouldn't we want embedded linux users to take advantage of this code?

Reply:
The design decision for OTA library is to use user allocated buffers but if the buffer is NULL/Zero then dynamic allocation si done.

Reply:
The design decision for OTA library is to use buffers passed by ota demo/application and when the buffer is NULL/zero then only use dynamic allocation. As this is in the POSIX OS port functions which are passed to the OTA library and user can override it to not use dynamic allocation at all.